src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -191,6 +192,11 @@ public void claimTask(String taskId,
             );
             //Lock & update Task
             TaskResource task = findByIdAndObtainLock(taskId);
+            if (task.getState() == CFTTaskState.ASSIGNED && !task.getAssignee().equals(userId)) {
+                cftTaskDatabaseService.saveTask(task);
```

Comment:
Do you need to save the task here? as you are throwing and exception, transaction will be rollback anyway. isn't it?